tests/databaseDiff.js

```diff
@@ -0,0 +1,270 @@
+const ERR = require('async-stacktrace');
+const async = require('async');
+const fs = require('fs-extra');
+const path = require('path');
+const pg = require('pg');
+const pgArray = require('pg').types.arrayParser;
+const assert = require('chai').assert;
+const colors = require('colors');
+const _ = require('lodash');
+const jsdiff = require('diff');
+
+const sqldb = require('../lib/sqldb');
+const sqlLoader = require('../lib/sql-loader');
+const describe = require('../tests/databaseDescribe');
+
+module.exports = {};
+
+function diff(db1, db2, options, callback) {
+    function formatText(text, formatter) {
+        if (options.coloredOutput && formatter) {
+            return formatter(text);
+        }
+        return text;
+    }
+
+    let db1Name = db1.type === 'database' ? db1.name : db1.path;
+    let db2Name = db2.type === 'database' ? db2.name : db2.path;
+
+    let description1, description2;
+    let result = '';
+
+    async.series([
+        (callback) => {
+            loadDescription(db1, (err, description) => {
+                if (ERR(err, callback)) return;
+                description1 = description;
+                callback(null);
+            });
+        },
+        (callback) => {
+            loadDescription(db2, (err, description) => {
+                if (ERR(err, callback)) return;
+                description2 = description;
+                callback(null);
+            });
+        },
+        (callback) => {
+            // Determine if both databases have the same tables
+            let missingFrom1 = _.difference(_.keys(description2.tables), _.keys(description1.tables));
+            let missingFrom2 = _.difference(_.keys(description1.tables), _.keys(description2.tables));
+            let db2NameBold = formatText(db2Name, colors.bold);
+
+            if (missingFrom1.length > 0) {
+                result += formatText(`Tables added to ${db2NameBold} (${db2.type})\n`, colors.underline);
+                result += formatText(missingFrom1.map(table => `+ ${table}`).join('\n') + '\n\n', colors.green);
+            }
+
+            if (missingFrom2.length > 0) {
+                result += formatText(`Tables missing from ${db2NameBold} (${db2.type})\n`, colors.underline);
+                result += formatText(missingFrom2.map(table => `- ${table}`).join('\n') + '\n\n', colors.red);
+            }
+
+            callback(null);
+        },
+        (callback) => {
+            // Determine if both databases have the same enums
+            let missingFrom1 = _.difference(_.keys(description2.enums), _.keys(description1.enums));
+            let missingFrom2 = _.difference(_.keys(description1.enums), _.keys(description2.enums));
+
+            if (missingFrom1.length > 0) {
+                let db1NameBold = formatText(db1Name, colors.bold);
+                result += formatText(`Enums added to ${db2NameBold} (${db1.type})\n`, colors.underline);
```

Comment:
With how I did this, we actually want db2 here. I made the diff show all changes in db2 relative to db1.